src/backend/distributed/master/master_create_shards.c

```diff
@@ -169,8 +170,14 @@ CreateShardsWithRoundRobinPolicy(Oid distributedTableId, int32 shardCount,
 	workerNodeList = ActivePrimaryNodeList();
 	workerNodeList = SortList(workerNodeList, CompareWorkerNodes);
 
-	/* make sure we don't process cancel signals until all shards are created */
-	HOLD_INTERRUPTS();
+	/*
+	 * Make sure we don't process cancel signals until all shards
+	 * are created if the executor is not enabled.
+	 */
+	if (DEFAULT_POOL_SIZE == 0)
```

Comment:
@pykello The PR is ready for review now. I tried hard to prevent adding new output files. But, create distributed table used to ignore the cancellations. Now, it should not. Even if I do not add this check, the executor would be processing the cancels and we'd still have various output differences.

So, I think do it make is a lot more accurate and useful to allow cancellations here.